### PR TITLE
[jquery] Adding more tests

### PR DESCRIPTION
**Pull request by HolgerJeromin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Added some tests from my closed PR #26027 and optimized other test comment. The test was copy and pasted from my comment and not easy to understand in this new context.

### CODE REVIEW

File: types/jquery/jquery-tests.ts [hunk 1]
```diff
@@ -35,12 +35,36 @@ function JQueryStatic() {
 
         // $ExpectType JQuery<HTMLElement>
         $('<p></p>');
+        
+        // $ExpectType JQuery<HTMLParagraphElement>
+        $<HTMLParagraphElement>('<p></p>');
```
Comment (leonard-thieu) [inline types/jquery/jquery-tests.ts]: There isn't a signature that supports this. I would remove the tests that declare the type parameter inline.

The alternative is to use a type assertion:
```typescript
const $p = $('<p></p>') as JQuery<HTMLParagraphElement>
// $ExpectType JQuery<HTMLParagraphElement>
$p;
```

or create a factory variable for that type:
```typescript
const $$p = $ as JQueryStatic<HTMLParagraphElement>;
// $ExpectType JQuery<HTMLParagraphElement>
$$p('<p></p>');
```

File: types/jquery/jquery-tests.ts [hunk 2]
```diff
@@ -35,12 +35,36 @@ function JQueryStatic() {
...
+        // $ExpectType JQuery<HTMLParagraphElement>
+        $<HTMLParagraphElement>('p', document);        
 
         // $ExpectType JQuery<HTMLElement>
         $(new HTMLElement());
 
         // $ExpectType JQuery<HTMLElement>
         $([new HTMLElement()]);
+        
+        // $ExpectType JQuery<HTMLElement>
```
Comment (leonard-thieu) [inline types/jquery/jquery-tests.ts]: Should be `// $ExpectType JQuery<HTMLParagraphElement>`.

**Review by leonard-thieu — CHANGES_REQUESTED** [review]
I'm debating whether it's appropriate to have a type parameter on `JQueryStatic` at all. Using it for anything outside of the default type requires awkward type assertions. It might be better served by individual type parameters on the APIs that currently use the type parameter. It'd be a breaking change for some consumers though.

File: types/jquery/jquery-tests.ts (hunk 2 above)
Comment (HolgerJeromin) [reply types/jquery/jquery-tests.ts]: Thanks, overlooked this type.

**Comment by HolgerJeromin** [comment]
> I'm debating whether it's appropriate to have a type parameter on JQueryStatic at all. Using it for anything outside of the default type requires awkward type assertions. 

This change was my motivation for the PR. But handling it correct is a huge task, you are right. Perhaps this is not worth the effort...

> It might be better served by individual type parameters on the APIs that currently use the type parameter. It'd be a breaking change for some consumers though.

I am not sure what you mean here.

**Comment by leonard-thieu** [comment]
> > It might be better served by individual type parameters on the APIs that currently use the type parameter. It'd be a breaking change for some consumers though.
> 
> I am not sure what you mean here.

I'm suggesting removal of `TElement` from `JQueryStatic`. APIs that were using `TElement` now need a replacement, so something like

```typescript
(selector_object_callback: JQuery.Selector | JQuery.htmlString | ((this: Document, $: JQueryStatic<TElement>) => void)): JQuery<TElement>;
```

would turn into

```typescript
<T = HTMLElement>(selector_object_callback: JQuery.Selector | JQuery.htmlString | ((this: Document, $: JQueryStatic<T>) => void)): JQuery<T>;
```

This would allow you to use `$<HTMLParagraphElement>('p')`.

---

The APIs using `TElement` aren't really related. Because of this, `TElement` doesn't improve type safety.

**Review by leonard-thieu — APPROVED** [review]

**Comment by sandersn** [comment]
@HolgerJeromin do you want me to merge this in its current state? All it does is remove some comments and add one `ExpectType` that will never fail.

```ts
let myWindowForced: JQuery<Window> = $(window);
// $ExpectType JQuery<Window>
myWindowForced;
```

myWindowForced will always have its declared type, so ExpectType won't fail there. You'll get the error on the `let myWindowForced: JQuery<Window>` line if it appears anywhere.

**Comment by HolgerJeromin** [comment]
@sandersn 
For this PR, yes. 
Not the big win I started the PR for, but i leave the decision about types at jquerystatic to @leonard-thieu in the future.

**Comment by HolgerJeromin** [comment]
@leonard-thieu 

> The alternative is to use a type assertion:
> `const $p = $('<p></p>') as JQuery<HTMLParagraphElement>`

How to I model a jquery collection of svg elements?

```ts
// works
let jSvgSvgElement = $(document.createElementNS("http://www.w3.org/2000/svg", "svg")); 

// JQuery<HTMLElement> can not be converted to JQuery<SVGLineElement>
let jSvgLineElement1 = $('.mysvgline') as JQuery<SVGLineElement>;
let jSvgLineElement2: JQuery<SVGLineElement> = $('.mysvgline');
```

**Comment by leonard-thieu** [comment]
Looks like you'll need to use a double assertion for now.

```typescript
let jSvgLineElement1 = $('.mysvgline') as any as JQuery<SVGLineElement>;
let jSvgLineElement2: JQuery<SVGLineElement> = $('.mysvgline') as any;
```

I know it looks ugly. Use cases like these probably warrants a closer look at the whole type parameter situation.

**Pull request merged**